core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/ScoreOperationChainHandler.java

```diff
@@ -0,0 +1,265 @@
+        loadMapsFromProperties(readProperties(operationScores), readProperties(authScores));
+    }
+
+    private void loadMapsFromProperties(final InputStream operationScores, final InputStream authScores) {
+        try {
+            loadMapsFromProperties(readProperties(operationScores), readProperties(authScores));
+        } finally {
+            CloseableUtil.close(operationScores);
+            CloseableUtil.close(authScores);
+        }
+    }
+
+    private void loadMapsFromProperties(final Properties operationScoreProperties, final Properties operationAuthScoreLimitProperties) {
+        Map<Class<? extends Operation>, Integer> opScores = new LinkedHashMap<>();
+        for (final String opClassName : operationScoreProperties.stringPropertyNames()) {
+            final Class<? extends Operation> opClass;
+            try {
+                opClass = Class.forName(opClassName)
+                        .asSubclass(Operation.class);
+            } catch (final ClassNotFoundException e) {
+                LOGGER.error("An operation class could not be found for operation score property {}", opClassName, e);
+                throw new IllegalArgumentException(e);
+            }
+            final Integer score = Integer.parseInt(operationScoreProperties.getProperty(opClassName));
+            opScores.put(opClass, score);
+        }
+        operationScores.putAll(sortByValue(opScores));
+        Map<String, Integer> authScores = new HashMap<>();
+        for (final String authName : operationAuthScoreLimitProperties
+                .stringPropertyNames()) {
+            final Integer score = Integer.parseInt(operationAuthScoreLimitProperties
+                    .getProperty(authName));
+            authScores.put(authName, score);
+        }
+        this.authScores.putAll(authScores);
+    }
+
+
+    private static Properties readProperties(final Path propFileLocation) {
+        Properties props;
+        if (null != propFileLocation) {
+            try {
+                props = readProperties(Files.newInputStream(propFileLocation, StandardOpenOption.READ));
+            } catch (final IOException e) {
+                throw new IllegalArgumentException(e);
+            }
+        } else {
+            props = new Properties();
+        }
+
+        return props;
+    }
+
+    private static Properties readProperties(final InputStream stream) {
+        final Properties props = new Properties();
+        if (null != stream) {
+            try {
+                props.load(stream);
+            } catch (final IOException e) {
+                throw new IllegalArgumentException("Failed to load store properties file : " + e
+                        .getMessage(), e);
+            } finally {
+                CloseableUtil.close(stream);
+            }
+        }
+        return props;
+    }
+
+    private static <K, V extends Comparable<? super V>> Map<K, V> sortByValue(final Map<K, V> map) {
```

Comment:
Good question. Both are a complex, so I don't think it is worth making the change.